app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt

```diff
@@ -322,6 +366,12 @@ class StateFragmentPresenter @Inject constructor(
 
     val ephemeralState = result.getOrThrow()
     explorationCheckpointState = ephemeralState.checkpointState
+    explorationCheckpoint = ephemeralState.explorationCheckpoint
+
+    if (hasExplorationJustLoaded) {
+      recyclerViewAssembler.restoreHintHandlerFromCheckpoint(explorationCheckpoint)
+    }
+
```

Comment:
here in the line ```explorationCheckpoint = ephemeralState.explorationCheckpoint``` I am initializing the variable `explorationCheckpoint` with the latest checkpoint, this variable will be saved in saved instance state in the override method if the configuration is changed so that it can be restored when the fragment is recreated.

Reply:
resolving this because changed the approach to handle hints